producer/src/main/java/com/internship/producer/service/ProductService.java

```diff
@@ -0,0 +1,36 @@
+package com.internship.producer.service;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.kafka.clients.admin.NewTopic;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.kafka.core.KafkaTemplate;
+import org.springframework.stereotype.Service;
+import store.bean.Product;
+
+@Service
+public class ProductService {
+
+    private static final String TOPIC = "test_topic";
+
+    private final KafkaTemplate<String, String> kafkaTemplate;
+
+    @Autowired
```

Comment:
You can use @RequiredArgsConstructor on top of the class, thereby providing dependency injection through the constructor.